src/bindings/c/include/openvino/c/ov_infer_request.h

```diff
@@ -0,0 +1,167 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for the ov_infer_request C API, which is a C wrapper for ov::InferRequest class
+ * This is a class of infer request that can be run in asynchronous or synchronous manners.
+ * @file ov_infer_request.h
+ */
+
+#pragma once
+
+#include "openvino/c/ov_common.h"
+#include "openvino/c/ov_tensor.h"
+
+typedef struct ov_infer_request ov_infer_request_t;
+
+/**
+ * @struct ov_callback_t
+ * @brief Completion callback definition about the function and args
+ */
+typedef struct {
+    void(OPENVINO_C_API_CALLBACK* callback_func)(void* args);
+    void* args;
+} ov_callback_t;
+
+/**
+ * @struct ov_ProfilingInfo_t
+ * @brief Store profiling info data
+ */
+typedef struct {
+    enum Status {       //!< Defines the general status of a node.
+        NOT_RUN,        //!< A node is not executed.
+        OPTIMIZED_OUT,  //!< A node is optimized out during graph optimization phase.
+        EXECUTED        //!< A node is executed.
+    } status;
+    int64_t real_time;      //!< The absolute time, in microseconds, that the node ran (in total).
+    int64_t cpu_time;       //!< The net host CPU time that the node ran.
+    const char* node_name;  //!< Name of a node.
+    const char* exec_type;  //!< Execution type of a unit.
+    const char* node_type;  //!< Node type.
+} ov_profiling_info_t;
+
+/**
+ * @struct ov_profiling_info_list_t
+ * @brief A list of profiling info data
+ */
+typedef struct {
+    ov_profiling_info_t* profiling_infos;
+    size_t size;
+} ov_profiling_info_list_t;
+
+// infer_request
+/**
+ * @defgroup infer_request infer_request
+ * @ingroup openvino_c
+ * Set of functions representing of infer_request.
+ * @{
+ */
+
+/**
+ * @brief Sets an input/output tensor to infer on.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param tensor_name  Name of the input or output tensor.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_tensor(ov_infer_request_t* infer_request, const char* tensor_name, const ov_tensor_t* tensor);
+
+/**
+ * @brief Sets an input tensor to infer on.
+ * @ingroup infer_request
+ * @param infer_request A pointer to the ov_infer_request_t.
+ * @param idx Index of the input tensor. If @p idx is greater than the number of model inputs, an exception is thrown.
+ * @param tensor Reference to the tensor.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_infer_request_set_input_tensor(ov_infer_request_t* infer_request, size_t idx, const ov_tensor_t* tensor);
```

Comment:
what's about get inout tensor by index

Reply:
Actually, we didn't wrap all C++ api in the first version, because it will be a huge number api from C++ to C mapping. We just map the APIs that are called in C sample so that can create a basic C inference task.